drools-ansible-rulebook-integration-api/src/main/java/org/drools/ansible/rulebook/integration/api/rulesengine/RulesSetEventStructure.java

```diff
@@ -0,0 +1,424 @@
+package org.drools.ansible.rulebook.integration.api.rulesengine;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.text.similarity.LevenshteinDistance;
+import org.drools.ansible.rulebook.integration.api.domain.Rule;
+import org.drools.ansible.rulebook.integration.api.domain.RuleContainer;
+import org.drools.ansible.rulebook.integration.api.domain.RulesSet;
+import org.drools.ansible.rulebook.integration.api.domain.conditions.MapCondition;
+import org.kie.api.runtime.rule.Match;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static java.util.stream.Collectors.toSet;
+
+/**
+ * This class holds the event paths from the rules set.
+ * Then, when the first event comes, validate the event paths comparing with incoming event structure.
+ * If the event path doesn't match with the incoming event structure, suggest the possible typo or missing node.
+ */
+public class RulesSetEventStructure {
+
+    private static final Logger LOG = LoggerFactory.getLogger(RulesSetEventStructure.class);
+
+    public static final String EVENT_STRUCTURE_SUGGESTION_ENABLED_ENV_NAME = "DROOLS_SUGGESTION_ENABLED";
```

Comment:
"DROOLS_SUGGESTION_ENABLED" is the env var for ansible-rulebook. We can enable this feature with
```
DROOLS_SUGGESTION_ENABLED=true ansible-rulebook ...
```